### DISCUSSION TITLE
Setting `application_name` by default for custom PostgreSQL driver implementations

### DISCUSSION THREAD

**Discussion by bobbyiliev** [dbeaver/dbeaver#20825] [DBeaver Development] [answered]
I've noticed that with all of the custom PostgreSQL drivers, like CockroachDB and Materialize for example, the `application_name` connection parameter defaults to `DBeaver 23.1.4 - Main <postgres>`.

Does anyone know if there is a way to change this in the drivers so that the name of the driver is also included by default without having to override the `openConnection` for example? e.g.  `Materialize DBeaver 23.1.4 - Main <postgres>`.

**Comment by E1izabeth** [comment]
You could override it here. And you can use variables from the tooltip to have a pattern like the default (DBeaver 23.1.4 - Main <postgres>).
![image](https://github.com/dbeaver/dbeaver/assets/28875055/223ecf43-c851-4b0d-9e1a-ae47294129c6)

**Comment by E1izabeth** [comment, marked as answer]
As for the code I believe there is no other way than overriding `openConnection`. Sometimes the driver wants the application identifier to be in a concrete form in some specific driver property, then it could be solved by overriding `getAllConnectionProperties` method for example like [here ](https://github.com/dbeaver/dbeaver/pull/20411/files#diff-27e3393ff8ed527d297a44b4bbece425e634612bedfa26d62d5568f55c5f4c1aR79)
